core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/ProjectClasspathChangedListener.java

```diff
@@ -49,24 +51,28 @@ public ProjectClasspathChangedListener() {
 	
 	@Override
 	public void elementChanged(ElementChangedEvent event) {
-		visit(event.getDelta());
+		if (isActivated) {
+			visit(event.getDelta());
+		}
 	}
 
 	@Override
     public void resourceChanged(IResourceChangeEvent event) {
-        IResource rsrc = event.getResource();
-        if (rsrc instanceof IProject) {
-        	IProject prj = (IProject)rsrc;
-            if (event.getType() == IResourceChangeEvent.PRE_CLOSE || 
-            	event.getType() == IResourceChangeEvent.PRE_DELETE) {
-            	// closed and deleted projects are not considered
-            	knownProjects.remove(prj);
-            } else {
-            	if (!knownProjects.containsKey(prj)) {
-            		initializeProject(prj);
-            	}
-            }            	
-        }
+		if (isActivated) {
+			IResource rsrc = event.getResource();
+			if (rsrc instanceof IProject) {
+				IProject prj = (IProject)rsrc;
+				if (event.getType() == IResourceChangeEvent.PRE_CLOSE || 
+						event.getType() == IResourceChangeEvent.PRE_DELETE) {
```

Comment:
Is it really a good idea to NOT remove the projects on delete/close even if the listener is deactived atm?

Reply:
yeah maybe better to put the if(isActivated) only in the Else